src/base/UGraphic.pas

```diff
@@ -554,43 +557,79 @@ procedure InitializeScreen;
   end; // case
 
   // If there is a resolution in Parameters, use it, else use the Ini value
-  I := Params.Resolution;
-  if (I <> -1) then
-    S := IResolution[I]
+  // check for a custom resolution (in the format of WIDTHxHEIGHT) or try validating ID from TIni
+  if ParseResolutionString(Params.CustomResolution, W, H) then
+    Log.LogStatus(Format('Use custom resolution from Command line: %d x %d', [W, H]), 'SDL_SetVideoMode')
+  else if Ini.GetResolution(Params.Resolution, S) and ParseResolutionString(S, W, H) then
+    Log.LogStatus(Format('Use resolution by index from command line: %d x %d [%d]', [W, H, Params.Resolution]), 'SDL_SetVideoMode')
   else
-    S := IResolution[Ini.Resolution];
+  begin
+    Log.LogStatus('Use config resolution', 'SDL_SetVideoMode');
+    S := Ini.GetResolution(W, H);
+  end;
 
-  I := Pos('x', S);
-  W := StrToInt(Copy(S, 1, I-1));
-  H := StrToInt(Copy(S, I+1, 1000));
   if ((Screens > 1) and not Split) then
   	W := W * Screens;
 
-  Log.LogStatus('SDL_SetVideoMode', 'Initialize3D');
+  Log.LogStatus('Creating window', 'SDL_SetVideoMode');
 
   // check whether to start in fullscreen or windowed mode.
   // The command-line parameters take precedence over the ini settings.
   Fullscreen := ((Ini.FullScreen = 1) or (Params.ScreenMode = scmFullscreen)) and
                 not (Params.ScreenMode = scmWindowed);
 
+  // TODO: use SDL renderer (for proper scale in "real fullscreen"). Able to choose rendering mode (OpenGL, OpenGL ES, Direct3D)
```

Comment:
switching from old OpenGL 1.x functions to all SDL functions will be a huge improvement and is necessary for better support of low performance hardware like RaspberryPi + later on support for android based devices. But this will be quite a lot of work. How about we do this piece by piece and start with moving from surfaces to textures and then somehow switch the Lyrics-rendering to non-OpenGl/glew functions?